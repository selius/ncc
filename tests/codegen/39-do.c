int main()
{
	int i;
	int a;

	i = 0;
	do {
		__print_int(i);
		++i;
	} while (i < 5);

	i = 0;
	do {
		++i;
	} while (0);
	__print_int(i);

	i = 0;
	do {
		++i;
		if (i > 999) {
			break;
		}
	} while (1);
	__print_int(i);

	a = 0;
	i = -1;
	do {
		++i;
		a += i;
		continue;
		a *= i;
	} while (i < 9);
	__print_int(a);
	
	return 0;
}
