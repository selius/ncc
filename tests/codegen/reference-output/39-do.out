0
1
2
3
4
1
1000
45
